ocaml/xapi/xapi_pool.ml

```diff
@@ -615,6 +615,44 @@ let protect_exn f x =
 		Debug.log_backtrace e (Backtrace.get e);
 		None
 
+let create_or_get_pvs_server_on_master __context rpc session_id pvs_site_ref (pvs_server_ref, pvs_server) : API.ref_PVS_server =
+	let my_uuid = pvs_server.API.pVS_server_uuid in
+	let new_pvs_server_ref =
+		try Client.PVS_server.get_by_uuid ~rpc ~session_id ~uuid:my_uuid
```

Comment:
Comparing PVS servers by UUID is not the right thing to do. What if we found a matching PVS site on the pool, which contains a PVS server with identical IP addresses but a different UUID? Then you'll end up with duplicate information.

The safest thing to do it is to not attempt to merge the servers of a site that was matched on the pool with the ones from the joining host. Just distinguish two cases:

1. We've found a site on the pool, matched by name. Use it, and leave the servers.
2. We have not found a matching site on the pool. Create one with the same parameters as on the joining host, including the servers.